### ISSUE TITLE
inline constant tag name in JSX transformation

### ISSUE THREAD

**Issue by pionxzh** [pionxzh/wakaru#115]
### Description

Constant tag name should be inlined into the JSX tag to reduce the noise.

source: https://github.com/pionxzh/wakaru/issues/113#issuecomment-1946779728

### Input code

```typescript
const Name = "div";
const attrs = {id: "x"};
const Comp = React.createElement(Name, attrs);
```

### Expected behavior

```typescript
const attrs = {id: "x"};
const Comp = <div {...attrs} />;
```

### Actual behavior

```jsx
const Name = "div";
const attrs = { id: "x" };
const Comp = <Name {...attrs} />;
```

**Comment by 0xdevalias** [comment]
Not sure if it's a good idea; but you could also use unlined spread syntax like this:

```
const Comp = <div {...{ id: "x" }} />;
```

Though I wonder; would there be any reason to not expand it into something like this:

```
const Comp = <div id="x" />;
```

**Comment by pionxzh** [comment]
Yes. If we want to implement props inlining, then the redundant {...{ ... } } will be removed.

**Comment by 0xdevalias** [comment]
@pionxzh I believe this should be reopened as per:

> #115 still needs the props inlining to be considered "completed". But let's merge this first.
> 
> _Originally posted by @pionxzh in https://github.com/pionxzh/wakaru/issues/129#issuecomment-2102362186_

**Comment by pionxzh** [comment]
This is resolved by the Rust rewrite (V1).
